monasca_events_api/tests/test_stream_definitions.py

```diff
@@ -0,0 +1,401 @@
+        res.status = 0
+        streamsObj.on_get(self._generate_req(), res, None)
+        self.assertEqual(returnStreams, json.loads(res.body))
+        self.assertEqual(res.status, '200 OK')
+
+    @mock.patch(
+        'monasca_events_api.common.repositories.mysql.mysql_repository.mdb')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_expire_actions')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_fire_actions')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_description')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_name')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.StreamDefinitions._validate_stream_definition')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_json_msg_body')
+    @mock.patch(
+        'monasca_events_api.v2.common.helpers.validate_json_content_type')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    def test_on_post_integrity_error(
+            self,
+            validate,
+            tenantid,
+            json,
+            readjson,
+            streamvalid,
+            getname,
+            desc,
+            fireactions,
+            expire,
+            repo):
+        """POST method  failed due to integrity error"""
+        validate.return_value = True
+        repo.connect.side_effect = AlreadyExistsException()
+        fireactions.return_value = "fire_actions"
+        getname.return_value = "Test"
+        expire.return_value = "expire_actions"
+        desc.return_value = "Stream_Description"
+        readjson.return_value = {u'name': "Test",
+                                 u'id': "1",
+                                 u'description': "Stream_Description",
+                                 u'select': "select",
+                                 u'group_by': "group_by",
+                                 u'fire_criteria': "fire_criteria",
+                                 u'expiration': "expiration",
+                                 u'fire_actions': "fire_actions",
+                                 u'expire_actions': "expire_actions",
+                                 u'actions_enabled': u'true'}
+
+        tenantid.return_value = '0ab1ac0a-2867-402d'
+        streamvalid.return_value = True
+
+        streamsObj = StreamDefinitionsSubClass()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            streamsObj.on_post(self._generate_req(), res)
+            self.assertFalse(
+                1,
+                msg="DB Integrity Error, should fail but passed")
+        except Exception as e:
+            self.assertRaises(falcon.HTTPInternalServerError)
+            self.assertEqual(e.status, '500 Internal Server Error')
+
```

Comment:
Can you help out here as I did try to look but the code is returning  500 on already exists.